_posts/2021-11-02-choosing-dask-chunk-sizes.md

```diff
@@ -0,0 +1,157 @@
+If the foreman (the Dask scheduler) tells workers to go and fetch a single brick at a time, then bring each one to where the wall is being built, you can see how this is going to be very slow and inefficient! The workers are spending most of their time moving between the wall and the pile of bricks. Much less time is going towards doing the actual work of mortaring bricks onto the wall.
+
+Instead, we can do this in a smarter way. The foreman (Dask scheduler) can tell the workers to go and bring one full wheelbarrow load of bricks back each time. Now workers are spending much less time moving between the wall and the pile of bricks, and the wall will be finished much quicker.
+
+## Too big is also a problem
+
+If the Dask array chunks are too big, this is also bad. Why?
+Chunks that are too large are bad because then you are likely to run out of working memory.
+You may see out of memory errors happening, or you might see performance decrease substantially as data spills to disk.
+
+When too much data is loaded in memory on too few workers, Dask will try to spill data to disk instead of crashing.
+Spilling data to disk makes things run very slowly, because all the extra read/write operations to disk. Things don't just get a little bit slower, they get a LOT slower, so it's smart to watch out for this.
+
+To avoid data being spilled to disk, watch the **worker memory plot** on the dask dashboard.
+Orange bars are a warning you are close to the limit, and gray means data is being spilled to disk - not good!
+Take a look at the next section for tips on using the Dask dashboard.
+
+## Using the Dask dashboard
+
+If you're not very familiar with the Dask dashboard, or you just sometimes forget where to find certain dashboard plots (like the worker memory plot), then you'll probably enjoy these quick video tutorials:
+
+- [Intro to the Dask dashboard (18 minute video)](https://www.youtube.com/watch?v=N_GqzcuGLCY)
+- [Dask Jupyterlab extension (6 minute video)](https://www.youtube.com/watch?v=EX_voquHdk0)
+- [Dask dashboard documentation](http://distributed.dask.org/en/latest/diagnosing-performance.html)
+
+We recommend always having the dashboard up when you're working with Dask.
+It's a fantastic way to get a sense of what's working well, or poorly, so you can make adjustments.
+
+Bad signs to watch out for include:
+- Lots of white space in the task stream plot is a bad sign. White space means nothing is happening. Chunks may be too small.
+- Lots and lots of red in the task stream plot is a bad sign. Red means worker communication. Dask workers need some communication, but if they are doing almost nothing except communication then there is not much productive work going on.
+- On the worker memory plot, watch out for orange bars which are a sign you are getting close to the memory limit. Chunks may be too big.
+- On the worker memory plot, watch out for grey bars which mean data is being spilled to disk. Chunks may be too big.
```

Comment:
Maybe sum-up the last two points as already explained in previous chapter?

Reply:
Probably better to keep this and remove the other one. We can link to this from that other section.

Reply:
Upon further review, my opinion is the same as above. Now that we have rearranged the order of sections, I think the small amount of repetition here is helpful to reinforce the key points.